controlplane/kubeadm/internal/machinefilters/machine_filters.go

```diff
@@ -197,3 +196,20 @@ func ControlPlaneSelectorForCluster(clusterName string) labels.Selector {
 		must(labels.NewRequirement(clusterv1.MachineControlPlaneLabelName, selection.Exists, []string{})),
 	)
 }
+
+// NeedsRemediation returns whether the machine has the
+// MachineOwnerRemediatedCondition set to false.
+func NeedsRemediation(m *clusterv1.Machine) bool {
+	return conditions.IsFalse(m, clusterv1.MachineOwnerRemediatedCondition)
```

Comment:
Do we need to check if the condition exists first: 
`conditions.Get(&m, clusterv1.MachineOwnerRemediatedCondition) != nil`